cypress/support/fragments/circulation-log/searchPane.js

```diff
@@ -40,22 +52,35 @@ export default {
     const dateRegEx = /\d{1,2}\/\d{1,2}\/\d{4},\s\d{1,2}:\d{2}\s\w{2}/gm;
 
     return cy.get('#circulation-log-list').then(element => {
-      const resultCount = element.attr('data-total-count');
+      let resultCount = element.attr('data-total-count');
+
+      // only 30 records shows on every page
+      if (resultCount > 29) {
+        resultCount = 29;
+      }
 
       // verify every string in result table
       for (let i = 0; i < resultCount; i++) {
         const resultRow = getResultRowByRowNumber(i);
         cy.do([
-          resultRow.userBarcode.perform(el => expect(el.textContent).to.match(/\d/)),
```

Comment:
Do you really need " cy.do([resultRow.userBarcode.perform(el => expect(el.textContent).to.match(/\d|/))," instead of cy.expect(resultRow.userBarcode.......)? Check https://frontside.com/interactors/docs/matchers please